src/Sub_monomial.jl

```diff
@@ -1,3 +1,112 @@
+function sub_monomial(
+                      mon1::Number,
+                      mon2::MMonomialLike{C},
+                      mon3::MPolynomialLike{C, T}) where {C,T}
+    return mon1
+end
+
+function sub_monomial(
+                      mon1::PolyVar{C},
+                      mon2::MMonomialLike{C},
+                      mon3::MPolynomialLike{C, T}) where {C,T}
+    if mon1 == mon2
+        return mon3
+    else
+        return mon1
+    end
+end
+
+function sub_monomial(
+                      mon1::Monomial{C},
+                      mon2::MMonomialLike{C},
+                      mon3::T) where {C, T <: Number}
+    mon1 = _reduce(mon1)
+    mon2 = _reduce(mon2)
+    reduced = false
+    coef = one(T)
+    while !reduced
+        factors = split(mon1, mon2)
+        if factors isa Nothing
+            reduced = true
+        else
+            mon1 = _reduce(safe_multiplication(first(factors), last(factors)))
+            coef *= mon3
+        end
+    end
+    return coef*mon1
+end
+
+function sub_monomial(
+                      mon1::Monomial{C},
+                      mon2::MMonomialLike{C},
+                      mon3:: MMonomialLike{C}) where {C}
+    mon1 = _reduce(mon1)
+    mon2 = _reduce(mon2)
+    reduced = false
+    while !reduced
+        factors = split(mon1, mon2)
+        if factors isa Nothing
+            reduced = true
+        else
+            mon1 = _reduce(safe_multiplication(safe_multiplication(first(factors), mon3), last(factors)))
+        end
+    end
+    return mon1
+end
+
+function sub_monomial(
```

Comment:
I would made this recursive too combine it with the previous one.
So that  definining
`const MTermLike{C,T} = Union{MMonomialLike{C},Term{C,T}}`
we have a single
`function sub_monomial(
                      mon1::Monomial{C},
                      mon2::MMonomialLike{C},
                      mon3::MTermLike{C, T}) where {C, T}`